src/expertsystem/reaction/__init__.py

```diff
@@ -405,8 +411,10 @@ def find_solutions(
             match_external_edges(final_solutions)
         return Result(
             final_solutions,
-            final_result.not_executed_rules,
-            final_result.violated_rules,
+            final_result.not_executed_node_rules,
```

Comment:
We indeed have to split up `Result`. That (as you suggested), `find_solutions` returns a `StateTransition` (#325) and some `ViolatedRules` that bundles the edge and node rules (possibly also the `not_executed_rules`, not sure).

Reply:
We need an issue for this where we can discuss the various possibilities and problems related to this.

Reply:
I think #325 suffices, only need to reformulate it a bit